### PR TITLE
Aggregation module - faster unordered_map implementation

### PR DESCRIPTION
**Pull request by Slabimic**
Modified implementation of aggregator module now using unordered_map and is about 90% faster.

### CODE REVIEW

**Comment by Slabimic** [comment]
Vyřešeny konflikty, soubory modulu aktualizovány, vše ostatní ponecháno z aktuálního masteru.

**Comment by cejkato2** [comment]
@Slabimic There are some errors during the build: https://travis-ci.org/CESNET/Nemea-Modules/builds/362522929#L1850
It should be fixed before merging.

In addition, it seems to me that there are already merged commits in this PR.
As we learnt from the practice, the best practice is to develop in a special branch in own repo, create PR and then delete this branch. You probably worked on master, we (probably me) merded PR as a squash commit, however, you didn't synchronized your master to CESNET/master properly...
Please try to avoid this in the future, it'll make the things easier...

File: aggregator/aggregator.cpp [hunk 1]
```diff
@@ -440,6 +460,7 @@ int main(int argc, char **argv)
 {
    int ret;
    signed char opt;
+   storage.reserve(2097152);        // Reserve enough space for records without need of rehash()
```
[the remark this answers is not in the document]
Comment (Slabimic) [reply aggregator/aggregator.cpp]: Redesigned to use the value as macro.

File: aggregator/README.md [hunk 2]
```diff
@@ -5,7 +5,11 @@
 # Aggregator module - README
 
 ## Description
-This module serves for UniRec records aggregation processing. User has to specify parameters for processing including key fields and applied aggregation function. Module can work with 3 different timeout types (Active, Passive, Global) or their combination (Mixed = Active,Passive).
+This module serves for UniRec records aggregation processing. The module can aggregate UniRec records based on user-specified keys, aggregation functions and time interval. The input of this module is a (infinit) sequence of UniRec records. The output of this module is a sequence of aggregated UniRec records according to user settings.
+
+"User can specify aggregation functions by options listed [below](./README.md#module-specific-parameters), all options may be used repeatedly.**URFIELD** stands for name of the UniRec field.
```
[the remark this answers is not in the document]
Comment (Slabimic) [reply aggregator/README.md]: Space added into description.

File: scalar-aggregator/aggregator.c [hunk 3]
```diff
@@ -686,7 +685,7 @@ int main(int argc, char **argv)
       FREE_MODULE_INFO_STRUCT(MODULE_BASIC_INFO, MODULE_PARAMS);
       return 6;
    }
-
+
```
[the remark this answers is not in the document]
Comment (Slabimic) [reply scalar-aggregator/aggregator.c]: Maybe conflict resolving result by hitting the space key.

**Pull request merged**